doc/tools/docgram/common.edit_mlg

```diff
@@ -2271,6 +2095,322 @@ ltac2_expr: [
 | DELETE _ltac2_expr
 ]
 
+ssrfwdview: [
+| REPLACE "/" ast_closure_term ssrfwdview      (* SSR plugin *)
+| WITH LIST1 ( "/" ast_closure_term )  TAG SSR
+| DELETE "/" ast_closure_term      (* SSR plugin *)
+]
+
+ssrfwd: [
+| REPLACE ":" ast_closure_lterm ":=" ast_closure_lterm      (* SSR plugin *)
+| WITH OPT ( ":" ast_closure_lterm ) ":=" ast_closure_lterm      TAG SSR
+| DELETE ":=" ast_closure_lterm      (* SSR plugin *)
+]
+
+ssrsetfwd: [
+| REPLACE ":" ast_closure_lterm ":=" "{" ssrocc "}" cpattern      (* SSR plugin *)
+| WITH OPT ( ":" ast_closure_lterm ) ":=" [ "{" ssrocc "}" cpattern | lcpattern ]      TAG SSR
+| DELETE ":" ast_closure_lterm ":=" lcpattern      (* SSR plugin *)
+| DELETE ":=" "{" ssrocc "}" cpattern      (* SSR plugin *)
+| DELETE ":=" lcpattern      (* SSR plugin *)
+]
+
+
+(* per @gares *)
+ssrdgens: [
+| REPLACE ":" ssrgen ssrdgens_tl      (* SSR plugin *)
+| WITH ":" ssrgen OPT ( "/" ssrgen )      TAG SSR
+]
+
+ssrdgens_tl: [ | DELETENT ]
+
+ssrgen: [
+| REPLACE ssrdocc cpattern      (* SSR plugin *)
+| WITH cpattern LIST0 [ LIST1 ident | cpattern ]    TAG SSR
+| DELETE cpattern      (* SSR plugin *)
+]
+
+OPTINREF: [ ]  (* todo: move just before splice (though seems to be in an infinite loop there :-( *)
```

Comment:
No, this global edit form doesn't take a list of nonterminals.  It does the operation for all nonterminals that can be empty.  The local edit form (i.e. for a single production) is still useful in a couple places.  I removed some of the latter where they weren't needed.

While formerly infinite SPLICE loops are now bounded, we still need to be control when to apply OPTINREF.  I think all edit operations should be invoked explicitly in `common.edit_mlg` rather than unconditionally.